File: Data Structures & Algorithms/combination-target-sum/submission-1.cpp
class Solution {
public:

    void backtrack(vector<vector<int>>& ans, vector<int>& nums,vector<int>& u2 , int u2trash, int i){
        if(i>=nums.size()){
            return;
        }
        if(u2trash==0){
        ans.push_back(u2);
        return;}
        else if(u2trash<0){
        return;
        }
        u2.push_back(nums[i]);
        backtrack(ans,nums, u2, u2trash-nums[i], i);
        u2.pop_back();

        backtrack(ans, nums, u2, u2trash, i+1);

        return;

}

    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        
        vector<vector<int>> ans={};
        vector<int> u2;

        backtrack(ans, nums,u2, target, 0);
        return ans;
    }
};

File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        ad={}
        bd={}
        if(len(s)!=len(t)):
            return False
        
        
        for x in s:
            if x in ad:
                ad[x]+=1
            else:
                ad[x]=1

        for y in t:
            if y in bd:
                bd[y]+=1
            else:
                bd[y]=1

        for k in ad.keys():
            if k in bd.keys() and bd[k]==ad[k]:
                continue
            else:
                return False

        return True